core/src/main/java/org/bitcoinj/core/Block.java

```diff
@@ -253,18 +253,12 @@ protected void parse(ByteBuffer payload) throws BufferUnderflowException, Protoc
 
     public static Block createGenesis(NetworkParameters n) {
         Block genesisBlock = new Block(n, BLOCK_VERSION_GENESIS);
-        Transaction t = createGenesisTransaction(n, genesisTxInputScriptBytes, FIFTY_COINS, genesisTxScriptPubKeyBytes);
-        genesisBlock.addTransaction(t);
+        Transaction tx = Transaction.coinbase(n, genesisTxInputScriptBytes);
+        tx.addOutput(new TransactionOutput(tx, FIFTY_COINS, genesisTxScriptPubKeyBytes));
+        genesisBlock.addTransaction(tx);
```

Comment:
Well, you could also see this as preparation for the inevitable TransactionBuilder. (-:

But I'll have a look at a better API.

Reply:
I looked into having passed a list of outputs into `Transaction.coinbase()` but it doesn't make make any sense. We can construct Transaction in one go but then we need to mutate all outputs – because of the `parentTransaction`.

I think for now its best to see this method as part of the future `TransactionBuilder` (which of course will be mutable).

Reply:
I should add we're closer to having at least `TransactionInput` and `TransactionOutput` immutable, so we should prefer that to Transaction, which is a longer way off.

Reply:
Agreed. I didn't want to make extra work here, just thought we _might_ be able to make an atomic constructor.